### PR TITLE
get_flask_app() protected against too early invocation (#789)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
Addresses #789.

### CODE REVIEW

File: src/taipy/gui/gui.py [hunk 1]
```diff
@@ -1680,11 +1680,14 @@ def _register_data_accessor(self, data_accessor_class: t.Type[_DataAccessor]) ->
 
     def get_flask_app(self) -> Flask:
         """Get the internal Flask application.
+        - Cannot be called before the `(Gui.)run^` method has been called.
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py]: You MUST add an empty line in order to isolate the short doc.

=> """Get the internal Flask application.

This method must be called \*\*after\*\* `(Gui.)run^` method was invoked.
..."""

File: src/taipy/gui/gui.py [hunk 2]
```diff
@@ -1680,11 +1680,14 @@ def _register_data_accessor(self, data_accessor_class: t.Type[_DataAccessor]) ->
...
 
         Returns:
             The Flask instance used.
         """
-        return self._server.get_flask()
+        if hasattr(self, "_server"):
+            return self._server.get_flask()
+        raise RuntimeError("get_flask_app() cannot get called before run() has been called.")
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py]: "get_flask_app() cannot be invoked before run() has been called."

**Review by FabienLelaquais — APPROVED** [review]
Minor (yet important) change for doc generation.

**Review by dinhlongviolin1 — APPROVED** [review]
thx Fred

**Pull request merged**